builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1488,9 +1505,34 @@ func (vm *virtualMachine) setupVirtualMachine(c *govmomi.Client) error {
 	}
 
 	var resourcePool *object.ResourcePool
-	if vm.resourcePool == "" {
-		if vm.cluster == "" {
-			resourcePool, err = finder.DefaultResourcePool(context.TODO())
+	var hostSystem *object.HostSystem
+	if vm.hostSystem == "" {
```

Comment:
This appears to be repeated code, would it make sense to encapsulate this logic into an interface?